src/packages/version-check/src/index.ts

```diff
@@ -0,0 +1,291 @@
+      /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g;
+    const WRAP_WIDTH = Math.min(120, process.stdout.columns || 0);
+    const center = (str: string, width: number) => {
+      const mid = (width - visibleCharacterLength(str)) / 2;
+      if (mid < 0) return str;
+
+      const left = Math.floor(mid);
+      const right = Math.ceil(mid);
+      return " ".repeat(left) + str + " ".repeat(right);
+    };
+    const visibleCharacterLength = (str: string) => {
+      // if the string contains unicode characters we need to count them,
+      // destructuring the string to get the characters as codePOints
+      return [...str.replace(reAnsiEscapes, "")].length;
+    };
+
+    const line1 = chalk`New {hex("${TruffleColors.porsche}") ${upgradeType}} version of ${packageName} available! {hex("${TruffleColors.watermelon}") ${currentVersion}} ⇢ {hex("${TruffleColors.green}") ${latestVersion}} `;
+    const line2 = chalk`{hex("${TruffleColors.porsche}") Changelog:} {hex("${TruffleColors.turquoise}") https://github.com/trufflesuite/${packageName}/releases/v${latestVersion}}`;
+    const line3 = chalk`Run {hex("${TruffleColors.green}") npm install -g ${packageName}@${latestVersion}} to update!`;
+    const width =
+      Math.max(
+        visibleCharacterLength(line1),
+        visibleCharacterLength(line2),
+        visibleCharacterLength(line3)
+      ) + 4;
+    const wrapWidth = Math.max(width, WRAP_WIDTH);
+    const vPipe = chalk`{hex("${TruffleColors.yellow}") ║}`;
+    const hLines = "═".repeat(width);
+    const emptyLine = center(
+      vPipe + " ".repeat(width) + vPipe,
+      Math.max(width, wrapWidth)
+    );
+    const message = [""];
+    message.push(
+      chalk`{hex("${TruffleColors.yellow}") ${center(
+        "╔" + hLines + "╗",
+        wrapWidth
+      )}}`
+    );
+    message.push(emptyLine);
+    message.push(center(vPipe + center(line1, width) + vPipe, wrapWidth));
+    message.push(center(vPipe + center(line2, width) + vPipe, wrapWidth));
+    message.push(center(vPipe + center(line3, width) + vPipe, wrapWidth));
+    message.push(emptyLine);
+    message.push(
+      chalk`{hex("${TruffleColors.yellow}") ${center(
+        "╚" + hLines + "╝",
+        wrapWidth
+      )}}`
+    );
+    message.push("");
+    this._logger.log(message.join("\n"));
+    return true;
+  }
+
+  // This is called with --version and is displayed each time
+  getVersionMessage() {
+    const currentVersion = this._currentVersion;
+    const { latestVersion } = this._config;
+    if (this.detectSemverChange(currentVersion, latestVersion)) {
+      return `note: there is a new version available! ${currentVersion} -> ${latestVersion}`;
+    }
+    return "";
+  }
+
+  static get DEFAULTS(): { config: VersionCheckConfig } {
+    return {
+      config: {
+        packageName: "ganache",
+        enabled: false,
+        url: "https://version.trufflesuite.com",
+        ttl: 300, // http2session.setTimeout
```

Comment:
I don't think 300ms is enough time. Ping RTT to Australia is about 300ms from my house, so while we will be using a CDN, I don't think we'll have that much room, especially on the off chance that the server will need to do the full npm round trip as well.

This value should probably be in the order of seconds.

Reply:
:+1: https://github.com/trufflesuite/ganache/pull/3285/commits/5d7e151622dfc506b85d851341aac45de70c7939